sdk/storage/azure-storage-files-datalake/test/ut/datalake_sas_test.cpp

```diff
@@ -11,194 +11,314 @@
+        const Files::DataLake::DataLakePathClient& pathClient,
+        const std::string& sasToken)
+    {
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_NO_THROW(pathClient1.GetProperties());
+    }
+
+    void VerifyDataLakeSasNonRead(
+        const Files::DataLake::DataLakePathClient& pathClient,
+        const std::string& sasToken)
+    {
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_THROW(pathClient1.GetProperties(), StorageException);
+    }
+
+    void VerifyDataLakeSasWrite(
+        const Files::DataLake::DataLakeFileClient& pathClient,
+        const std::string& sasToken)
+    {
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_NO_THROW(pathClient1.UploadFrom(reinterpret_cast<const uint8_t*>("a"), 1));
+    }
+
+    void VerifyDataLakeSasDelete(
+        const Files::DataLake::DataLakeFileClient& pathClient,
+        const std::string& sasToken)
+    {
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_NO_THROW(pathClient1.Delete());
+      pathClient.UploadFrom(reinterpret_cast<const uint8_t*>("a"), 1);
+    }
+
+    void VerifyDataLakeSasDelete(
+        const Files::DataLake::DataLakeDirectoryClient& pathClient,
+        const std::string& sasToken)
+    {
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_NO_THROW(pathClient1.DeleteRecursive());
+      pathClient.Create();
+    }
+
+    void VerifyDataLakeSasCreate(
+        const Files::DataLake::DataLakeFileClient& pathClient,
+        const std::string& sasToken)
+    {
+      pathClient.DeleteIfExists();
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_NO_THROW(pathClient1.Create());
+    }
+
+    void VerifyDataLakeSasCreate(
+        const Files::DataLake::DataLakeDirectoryClient& pathClient,
+        const std::string& sasToken)
+    {
+      pathClient.DeleteRecursiveIfExists();
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_NO_THROW(pathClient1.Create());
+    }
+
+    void VerifyDataLakeSasList(
+        const Files::DataLake::DataLakeDirectoryClient& pathClient,
+        const std::string& sasToken)
+    {
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      EXPECT_NO_THROW(pathClient1.ListPaths(true));
+    }
+
+    void VerifyDataLakeSasMove(
+        const Files::DataLake::DataLakeDirectoryClient& pathClient,
+        const std::string& baseDirectoryName,
+        const std::string& sasToken)
+    {
+      auto pathClient1 = GetSasAuthenticatedClient(pathClient, sasToken);
+      std::string fileName = RandomString();
+      std::string newFilename = RandomString();
+      auto fileClient = pathClient.GetFileClient(fileName);
+      auto newFileClient = pathClient.GetFileClient(newFilename);
+      fileClient.Create();
+      newFileClient.Create();
```

Comment:
`fileClient` is source, `newFileClint` is destination right? Why do we need to create `newFileClient`?